raiden/network/transport/matrix/transport.py

```diff
@@ -738,19 +738,27 @@ def _initialize_first_sync(self) -> None:
         assert self._client.sync_thread is None, msg
         assert self._client.message_worker is None, msg
 
+        # search for broadcast room ids via public rooms list
+        broadcast_room_ids = self._client.public_room_ids(make_room_alias(self.chain_id, ""))
+
+        # TODO: Uncomment once the server updated their public room list
+        # assert len(broadcast_room_ids) >= len(self._config.broadcast_rooms)
+
         # Call sync to fetch the inventory rooms and new invites. At this point
         # the messages themselves should not be processed because the room
         # callbacks are not installed yet (this is done below). The sync limit
         # prevents fetching the messages.
-        filter_id = self._client.create_sync_filter(limit=0)
+        filter_id = self._client.create_sync_filter(
+            not_rooms=[Room(self._client, room_id) for room_id in broadcast_room_ids], limit=0
```

Comment:
Can't we rid of the first sync completely now?

Reply:
There is definitely taken a bit of need of why the first sync is needed manually. But there are still some reasons why we cannot go without it.
The filters are still different. During first sync we filter all messages of the rooms. This is due to the fact that we cannot process them before we assigned the room ids to the addresses. (Removing an attack vector)

The first sync is now mainly needed to only get the room_ids the user has joined and map these rooms to the partners. This needs to be done in advance before receiving messages by them. In theory we would not rely on the matrix server to store messages for us while we are offline since we require the other peer to retry after a while. But certainly this could cost some time especially if there are ordered messages in the pipeline to be processed. So this is mainly a performance gain.